tests/io/test_file.py

```diff
@@ -71,3 +72,47 @@ def test_open_file_write_zstd(self, tmpdir):
             f_write.write('foo\nbar\n')
         with zstd.open(path, 'rt') as f_read:
             assert f_read.read() == 'foo\nbar\n'
+
+    def test_open_file_nested_read(self, tmpdir):
+        """Write a text file compressed with zstd, then try to read it twice."""
+        path = str(tmpdir / 'test.txt')
+        with open(path, 'wt') as f_write:
+            f_write.write('foo\nbar\n')
+
+        with augur.io.file.open_file(path) as f_read:
+            with augur.io.file.open_file(f_read) as f_read:
+                assert f_read.read() == 'foo\nbar\n'
```

Comment:
_non-blocking_

A minor thing, but these are nested opens (in read mode), not nested reads.

```suggestion
    def test_open_file_nested_read(self, tmpdir):
        """Write a text file compressed with zstd, then try to open it as a path and then buffer for reading."""
        path = str(tmpdir / 'test.txt')
        with open(path, 'wt') as f_write:
            f_write.write('foo\nbar\n')

        with augur.io.file.open_file(path) as f_read:
            with augur.io.file.open_file(f_read) as f_read:
                assert f_read.read() == 'foo\nbar\n'
```

Ditto for the other functions below.

Reply:
Done in [force-pushes](https://github.com/nextstrain/augur/compare/b004bfe164f87a985ce3c7c55227ab3a6f2bbe72..205b70e03a8099a383d05addb2024a0b6e14e948).